src/main/java/nl/hsac/fitnesse/fixture/util/mail/ImapAttachment.java

```diff
@@ -0,0 +1,30 @@
+package nl.hsac.fitnesse.fixture.util.mail;
+
+import org.apache.commons.io.IOUtils;
+
+import javax.mail.MessagingException;
+import javax.mail.Part;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ImapAttachment {
+    private InputStream stream;
+    private String fileName;
+
+    public ImapAttachment(Part p) throws IOException, MessagingException {
+        this.stream = p.getInputStream();
+        this.fileName = p.getFileName();
+    }
+
+    public String getFileName() {
+        return this.fileName;
+    }
+
+    public InputStream getStream() {
+        return this.stream;
+    }
+
+    public byte[] getBytes() throws IOException  {
+        return IOUtils.toByteArray(this.stream);
```

Comment:
This way each attachment's content can only be accessed once, so one can get it by name of number, but not both. Is that intentional (that doesn't seem like a desirable feature to me)?
Otherwise this class should either store the part itself (so a new inputstream can be retrieved in each cal to `getBytes()`) or the byte array returned by the first call to `getBytes()` should be stored and reused in subsequent calls.

Or am I missing something here?

Reply:
In my test project it saves the same file/attachment fine multiple times

could `return IOUtils.toBufferedInputStream(this.stream);` is realy want though

Reply:
Good to hear that it already works. That must be because the input stream returned by the part already is some buffered version. Using the method you suggest does sound preferable, as that removes our dependency on implementation detail of the part's input stream.

Reply:
Your correct, fixed it